command/agent/job_endpoint.go

```diff
@@ -376,13 +382,20 @@ func (s *HTTPServer) jobUpdate(resp http.ResponseWriter, req *http.Request,
 
 	sJob := ApiJobToStructJob(args.Job)
 
+	// Backfill region from Job, if not present in WriteRequest
+	region := args.WriteRequest.Region
+	if region == "" {
+		region = *args.Job.Region
```

Comment:
as above, we should probably protect against region being nil, to avoid panicing a server on a malicious request